calendar/src/main/java/com/minuk/calendar/PetCalendarView.kt

```diff
@@ -0,0 +1,304 @@
+    }
+
+    /**
+     * 유클리드 호제법으로 구하는 a와 b의 최대공약
+     */
+    private fun getGCD(a: Int, b: Int): Int {
+        var maximum = max(a, b)
+        var minimum = min(a, b)
+
+        while (minimum != 0) {
+            val r = maximum % minimum
+
+            if (r == 0) return minimum
+
+            maximum = b
+            minimum = r
+        }
+
+        return minimum
+    }
+
+    fun setCalendarDate(
+        calendar: Calendar
+    ) {
+        setCalendarDate(
+            calendar[Calendar.YEAR],
+            calendar[Calendar.MONTH],
+            calendar[Calendar.DAY_OF_MONTH]
+        )
+    }
+
+    fun setCalendarDate(
+        year: Int = calendar[Calendar.YEAR],
+        month: Int = calendar[Calendar.MONTH],
+        day: Int = calendar[Calendar.DAY_OF_MONTH]
+    ) {
+        calendar.set(year, month, day)
+
+        initCalendarData()
+    }
+
+    fun setEventHandler(eventHandler: PetCalendarEventHandler) {
+        this.eventHandler = eventHandler
+    }
+
+    private inner class PetCalendarAdapter :
+        RecyclerView.Adapter<PetCalendarAdapter.PetCalenderViewHolder>() {
+        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): PetCalenderViewHolder {
+            val inflater = LayoutInflater.from(parent.context)
+            return PetCalenderViewHolder(ItemCalendarBinding.inflate(inflater))
+        }
+
+        override fun onBindViewHolder(holder: PetCalenderViewHolder, position: Int) {
+
+            val day: Int = getDayFromPosition(calendar[Calendar.MONTH], position).second
+
+            holder.binding.dayCalendarTv.text = "$day"
+
+            //현재 요일의 경우 동그라미 이미지 표시
+            if (position == toDay + startDayOfWeek - 1) {
+                holder.binding.dayCalendarTv.background =
+                    ResourcesCompat.getDrawable(resources, R.drawable.bg_date_selected, null)
+            } else {
+                holder.binding.dayCalendarTv.background = null
+            }
+
+
+            if (position == toDay + startDayOfWeek - 1) {
+                holder.binding.dayCalendarTv.setTextColor(todayColor)
+            } else if (position % 7 == 0 || position % 7 == 6) {
+                holder.binding.dayCalendarTv.setTextColor(weekendColor)
+            } else {
+                holder.binding.dayCalendarTv.setTextColor(weekdaysColor)
+            }
+        }
+
+        override fun getItemCount(): Int {
+            return daysCount
+        }
+
+        /**
+         * position 값으로 날짜 측정하는 함수
+         * 한 달력 화면에서 최대 3개의 월 달력이 보여 월과 일을 구분
+         *
+         * @return month 와 day 를 Pair 객체로 반환합니다.
+         */
+        private fun getDayFromPosition(
+            currentMonth: Int = calendar[Calendar.MONTH],
+            position: Int
+        ): Pair<Int, Int> = when {
+            position < startDayOfWeek -> {
+                val day = endDayOfLastMonth - (startDayOfWeek - position - 1)
+                Pair(currentMonth - 1, day)
```

Comment:
```suggestion
            position < startDayOfWeek -> currentMonth - 1 to (endDayOfLastMonth - (startDayOfWeek - position - 1))
```
이렇게 한 줄로 바꿀수 있을것 같아요!

Reply:
저는 `endDayOfLastMonth - (startDayOfWeek - position - 1) 수식이 day라는 것을 명시해주는 편이 가독성 측면에서 좋다고 생각했는데, 로직이 복잡하지 않아서 (단순 반환이라서) 코드를 줄이는 것이 더 좋은가요?

Reply:
그런 이유라면 좋습니다!